packages/react-i18n/src/i18n.ts

```diff
@@ -343,18 +344,25 @@ export class I18n {
     return WEEK_START_DAYS.get(country) || DEFAULT_WEEK_START_DAY;
   }
 
-  getCurrencySymbol = (currencyCode?: string) => {
+  getCurrencySymbol = (
```

Comment:
Something's been bugging me about the API design, and I haven't been able to think about why until now.

We want this API to be opinionated towards Shopify's Polaris guidelines on internationalization, and so `formatCurrency` is using `getShortCurrencySymbol` to get the short form currency.

However, we're exposing `getCurrencySymbol`, `getCurrencySymbolLocalized`, and `getShortCurrencySymbol`, and not making it clear that `getShortCurrencySymbol` is the recommended method.

I have a bit of a wild proposal:
- Update `getCurrencySymbolLocalized` to actually have the same logic as `getShortCurrencySymbol`, but allow the `options` parameter of type `Pick<Intl.NumberFormatOptions, 'currencyDisplay'>` to allow someone to override it if they want to.
- The `getCurrencySymbol` method remains as it is, also passing along the `options` parameter.
- We remove the `getShortCurrencySymbol`.

This would have an added benefit of automatically fixing the currency formatting for any code that is currently calling `getCurrencySymbol` or `getCurrencySymbolLocalized`.

This would end up ultimately negating the previous PR that just went out to expose `getShortCurrencySymbol`, but I think it might be better to update the methods we currently have than to add another option to the mix.

Thoughts?

Reply:
> We want this API to be opinionated towards Shopify's Polaris guidelines on internationalization,

Strong agree on this!

> This would end up ultimately negating the previous PR that just went out to expose getShortCurrencySymbol

We've merged the PR but not yet released it. If we revert it then we have no breaking change. I agree with improving our current functions over exposing multiple slightly different ones.

Yesterday I was thinking about if it'd be possible to remove getCurrencySymbolLocalized and make its behaviour be an option of getCurrencySymbol but I'm not sure that's possible.